smasher.md

```diff
@@ -0,0 +1,593 @@
+- associates the new job to the new Dataset
+- runs the smasher pipeline
+- asserts the job completed successfully
+- opens the `aggregated_metadata.json` file and asserts that the `quantile_normalized` field is set to True
+
+### `test_log2`
+
+Tests that an Experiment with non-log2 data is handled correctly
+
+test steps:
+- creates prerequisites
+    - creates test ProcessorJob
+    - creates test Experiment
+        - uses "GSE44421" which has non-log2 data
+    - creates test ComputationalResult
+    - creates test Organism
+    - creates test Sample
+    - associates the test Sample to the ComputationalResult and Experiment
+    - creates test ComputedFile and associates it to the test Sample
+    - creates second test Sample
+    - associates the second test Sample to the same ComputationalResult and Experiment
+    - creates second test ComputedFile and associates it to the second test Sample
+    - creates test Dataset using the two Samples
+    - associates the Dataset to the test ProcessorJob
+- runs the smasher pipeline
+- asserts that the job completed successfully
+
+### `test_dualtech_smash`
+
+Tests the smasher pipeline  on a Dataset with Samples that use both MICROARRAY and RNASEQ technologies.
+
+test steps:
+- creates prerequisites
+    - creates test ProcessorJob
+    - creates MICROARRAY prerequisites
+        - creates test Experiment
+        - creates test ComputationalResult
+        - creates test Organism
+        - creates test MICROARRAY Sample
+        - associates the test Sample to the ComputationalResult and Experiment
+        - creates test ComputedFile and associates it to the test Sample
+    - creates RNASEQ prerequisites
+        - creates test Experiment
+        - creates test ComputationalResult
+        - creates test Organism
+        - creates test RNASEQ Sample
+        - associates the test Sample to the ComputationalResult and Experiment
+        - creates test ComputedFile and associates it to the test Sample
+- Aggregate by "SPECIES"
+    - creates test Dataset using the two Experiments and Samples
+    - associates the test dataset to the ProcessorJob
+    - asserts that `dataset.is_cross_technology()` returns True
+    - runs the smasher pipeline
+    - asserts that the output file exists
+    - removes the output file
+    - asserts that the final output frame has 2 columns
+- aggregate by "EXPERIMENT"
+    - gets the test Dataset and sets `aggregate_by` to be "EXPERIMENT"
+    - resets the test ProcessorJob start and end times
+    - runs the smasher pipeline
+    - asserts that the output file exists
+    - removes the output file
+    - asserts that the final output frame has 1 columns
+- aggregate by "ALL"
+    - gets the test Dataset and sets `aggregate_by` to be "ALL"
+    - resets the test ProcessorJob start and end times
+    - runs the smasher pipeline
+    - asserts that the output file exists
+    - asserts that the final output frame has 1 columns
```

Comment:
For this and EXPERIMENT, shouldn't there still be two columns, just in different matrices?

Reply:
I was just wrong for "ALL" it should be 2.

I think for experiment, there would be 2 files, but `job_context["final_frame"]` just gets overwritten for each key that's smashed. So I think it's only checking for the second Experiment which would only have 1 column.